Feature/rdkemw 5775 final

rekhap2kandhavelan opened the pull request:
make middleare-player-interface L1 script file into aamp yml file

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: Migration to dev_sprint_25_2 from 117308 (patchset 12)
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: Migration to dev_sprint_25_2 from 117308 (patchset 12)
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

dp0000 committed:
Delete AampConfig.cpp.rej

removed .rej

dp0000 committed:
Delete aampgstplayer.cpp.rej

dp0000 committed:
Delete CMakeLists.txt.rej

Deepikasri1234 committed:
Merge branch 'feature/RDK-56050' of https://github.com/rdkcentral/aamp into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge remote-tracking branch 'origin/dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'develop' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Commit:
DISABLE_SECURITY_TOKEN secmanager support

rekhap2kandhavelan committed:
Update L1-tests.yml

rkandh015 committed:
Rebase with dev_sprint_25_2

rkandh015 committed:
rebase with dev_sprint

rekhap2kandhavelan committed:
Update L1-tests.yml

rekhap2kandhavelan committed:
Update L1-tests.yml

Commit:
MonitorAVState avposition data type change

rkandh015 committed:
rebase with dev_sprint

Change-Id: I77d0d29b44a01ae8d9fb3ddd455815700a3dbc76

narenr94 committed:
dev_sprint_25_2 sync

narenr94 committed:
Merge remote-tracking branch 'origin/dev_sprint_25_2' into feature/RDK-56050

narenr94 committed:
build error fix 03

rkandh015 committed:
Resolve L1 build failure for aamp and middleware

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
resolved compilation issues

rekhap2kandhavelan committed:
Update L1-tests.yml

Update yml file for build middleware player interface into aamp

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=308) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=308) before we can accept your contribution.<br/>**3** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: dp0000<br/>:white_check_mark: rekhap2kandhavelan<br/>:white_check_mark: narenr94<br/>:x: Naren Ramesh<br/>:x: Deepikasri1234<br/>:x: rkandh015<br/><hr/>**Naren Ramesh** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=308) it.</sub>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
84f7f23d6cde983b99d3800e05c2776c
PRESENT
-->

# Coverity Issue - Unchecked return value
Calling "wait_for" without checking return value (as is done elsewhere 13 out of 14 times).

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:2790](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L2790)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f729b73ba6db645adda7d67798226805
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"Identifier" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<GstPlayerPriv::CallbackData, std::allocator<GstPlayerPriv::CallbackData> >::push_back(std::vector<GstPlayerPriv::CallbackData, std::allocator<GstPlayerPriv::CallbackData> >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""Identifier"")" instead of "Identifier".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:1670](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L1670)

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -500,11 +536,12 @@ void InterfacePlayerRDK::SetPauseOnStartPlayback(bool enable)
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b65726fd04c9209310ef1e9427d61421
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1280,7 +1337,7 @@ void InterfacePlayerRDK::TearDownStream(GstMediaType mediaType)
-			gstPrivateContext->decoderHandleNotified = false;
+			interfacePlayerPriv->gstPrivateContext->decoderHandleNotified = false;
 		}
 		stream->format = GST_FORMAT_INVALID;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
cc3d03b3b37d92b0e947325f1455fe9d
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "format" to a new value. Now the two threads have an inconsistent view of "format" and updates to fields correlated with "format" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "format" and the read used to decide whether to modify "format" with the same set of locks.

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1786,6 +1851,7 @@ static GstPadProbeReturn InterfacePlayerRDK_DemuxPadProbeCallbackAny(GstPad *pad
 {
 	InterfacePlayerRDK* pInterfacePlayerRDK = (InterfacePlayerRDK *)_this;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
7568a269a0b816e3656db6fb94ff650d
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -549,12 +597,13 @@ static void GstPlayer_OnAudioFirstFrameAudDecoder(GstElement* object, guint arg0
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
dcb2bdfd0d07b3768b0c3b490224470e
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -56,29 +57,28 @@
+mSourceSetupCV(), mScheduler(), callbackMap(), setupStreamCallbackMap(), mDrmSystem(NULL), mEncrypt(NULL), mDRMSessionManager(NULL)
 {
-	gstPrivateContext = new GstPlayerPriv();
+	interfacePlayerPriv = new InterfacePlayerPriv();
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
69351cb2955e2120b33e31d236a4b44f
PRESENT
-->

# Coverity Issue - Resource leak in object
Allocating memory by calling "new InterfacePlayerPriv".

_High Impact, CWE-401_
_CTOR_DTOR_LEAK_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1393,30 +1450,30 @@ void InterfacePlayerRDK::Stop(bool keepLastFrame)
-	gstPrivateContext->videoMuted = false;
-	gstPrivateContext->subtitleMuted = false;
-	gstPrivateContext->audioVolume = 1.0;
+	interfacePlayerPriv->gstPrivateContext->audioMuted = false;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8df2dd462935a54e628498fbb12c7112
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->interfacePlayerPriv->gstPrivateContext->audioMuted" without holding lock "GstPlayerPriv.volumeMuteMutex". Elsewhere, "GstPlayerPriv.audioMuted" is written to with "GstPlayerPriv.volumeMuteMutex" held 2 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -743,10 +795,11 @@ gboolean InterfacePlayerRDK::IdleCallback(gpointer user_data)
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
0596248cc68cf99fdfe6af3569d8e6d5
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4863,28 +4961,16 @@ void InterfacePlayerRDK::EndOfStreamReached(int mediaType, bool &shouldHaltBuffe
-	retvalue = this->SetupStream(streamId, (void*)this, manifestUrl);
+	GstMediaType mediaType = static_cast<GstMediaType>(streamId);
+	this->TriggerEvent(InterfaceCB::startNewSubtitleStream, mediaType);
+	retvalue = this->SetupStream(mediaType, (void*)this, manifestUrl);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
900484ff8a70f30ce33dd9c73e270670
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"manifestUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""manifestUrl"")" instead of "manifestUrl".

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -694,14 +744,15 @@ void MonitorAV( InterfacePlayerRDK *pInterfacePlayerRDK )
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
2e4fa3aa18720f403b83acb5626db22a
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -714,22 +765,23 @@ gboolean InterfacePlayerRDK::ProgressCallbackOnTimeout(gpointer user_data)
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1eafa348f9641a42f2f5b89eeca20737
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4528,6 +4618,7 @@ bool InterfacePlayerRDK::SetTextStyle(const std::string &options)
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d5cf957472b0ed9788dac436016f06c1
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4385,18 +4473,19 @@ void type_check_instance(const char * str, GstElement * elem)
 	bool isPlayerReady = false;
 
-	if(pInterfacePlayerRDK && pInterfacePlayerRDK->gstPrivateContext)
+	if(pInterfacePlayerRDK && privatePlayer->gstPrivateContext)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d2b25a6a30b12f0b405946e0f4872c40
PRESENT
-->

# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design
Reason for change: comp issues
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

rkandh015 committed:
Merge branch 'feature/RDK-56050' into feature/RDKEMW-5775_final
